### DISCUSSION TITLE
Could not resolve "@prisma/client/runtime/index-browser"

### DISCUSSION THREAD

**Discussion by JakovGlavac** [prisma/prisma#10767] [Q&A] [answered]
Hi, while using Prisma with remix, I got this random error.

```
 > node_modules/.prisma/client/index-browser.js:6:12: error: Could not resolve "@prisma/client/runtime/index-browser" (mark 
it as external to exclude it from the bundle, or surround it with try/catch to handle the failure at run-time)
    6 │ } = require('@prisma/client/runtime/index-browser')
```

![image](https://user-images.githubusercontent.com/34971091/146689308-a67abcc6-5b1c-4914-af8f-20bc4c268b65.png)

 After I've investigated it further I found that the solution is to just comment out this import.
```ts
const {
  Decimal
} = require('@prisma/client/runtime/index-browser')
```
![image](https://user-images.githubusercontent.com/34971091/146689363-f2479888-082b-45f7-bb55-4b9a0344552b.png)

But I cant still figure out why would you import module inside itself, because file that it imports is that file.
![image](https://user-images.githubusercontent.com/34971091/146689398-6433211e-a537-458a-adf4-13162c3ab409.png)

I don't know if this is bug, or I did something wrong.

**Comment by janpio** [comment, marked as answer]
> But I cant still figure out why would you import module inside itself, because file that it imports is that file.

Small difference in the path - you are in `.prisma/client/index-browser.js` which tries to import `Decimal` from `@prisma/client/index-browser.js`. If that makes sense and why that is I can also not say though.

**Reply by Rykuno** [reply]
you have to run `yarn run prisma generate`.

**Comment by jycouet** [comment]
I have the same issue with SvelteKit.
I think that it's related with the bundler (can't really map `.prisma`).

Did you manage something?

**Reply by wvhulle** [reply]
> I have the same issue with SvelteKit. I think that it's related with the bundler (can't really map `.prisma`).
> 
> Did you manage something?

I posted a related issue to SvelteKit here on the  issue tracker. https://github.com/prisma/prisma/issues/12504

**Comment by harshmangalam** [comment]
I got the same error in qwikcity while using any edge adapter like vercel or netlify but working fine in node adapter

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by rksahu2610** [comment]
Loaded Prisma config from prisma.config.ts.

Prisma schema loaded from prisma\schema.prisma.

✔ Generated Prisma Client (7.2.0) to .\generated\prisma in 115ms

▲ Next.js 16.1.1 (Turbopack)
- Environments: .env
- Experiments (use with caution):
  · optimizePackageImports

  Creating an optimized production build ...

> Build error occurred
Error: Turbopack build failed with 1 errors:
./node_modules/@prisma/client/index-browser.js:1:16
Module not found: Can't resolve '.prisma/client/index-browser'
> 1 | const prisma = require('.prisma/client/index-browser')
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  2 |
  3 | module.exports = prisma
  4 |



Import trace:
  Client Component Browser:
    ./node_modules/@prisma/client/index-browser.js [Client Component Browser]
    ./src/hooks/use-auth.tsx [Client Component Browser]
    ./src/components/common/header.tsx [Client Component Browser]
    ./src/components/common/header.tsx [Server Component]
    ./src/app/(store)/layout.tsx [Server Component]

https://nextjs.org/docs/messages/module-not-found


    at ./node_modules/ (prisma/client/index-browser.js:1:16)
    at <unknown> (https://nextjs.org/docs/messages/module-not-found)
    
    i use `patch-package`
    
    - npx patch-package @prisma/client
    
    - patches/@prisma+client+7.2.0.patch
    
**Add this code**
```
    diff --git a/node_modules/@prisma/client/index-browser.js b/node_modules/@prisma/client/index-browser.js
index 3ea8d77..712bd50 100644
--- a/node_modules/@prisma/client/index-browser.js
+++ b/node_modules/@prisma/client/index-browser.js
@@ -1,3 +1,3 @@
-const prisma = require('.prisma/client/index-browser')
+const prisma = require('@prisma/client/index-browser')
 
 module.exports = prisma
```
